### PR TITLE
feat(LINGUIST-451) Active words improvements

### PR DESCRIPTION
**Pull request by selimcanglr**
- Fix active words modal not fetching active words properly
- Highlight active words on chat messages
- Signify active words on definition cards

### CODE REVIEW

File: components/chat/ActiveWordsModal.tsx [hunk 1]
```diff
@@ -5,44 +5,59 @@ import { useDispatch, useSelector } from 'react-redux';
...
   visible: boolean;
   setVisible: (visible: boolean) => void;
 }
 
 const ActiveWordsModal = ({ visible, setVisible }: ActiveWordsModalProps) => {
-  const { refetch } = useGetAllConversationsQuery();
-  const dispatch = useDispatch();
   const conversation = useSelector(selectCurrentConversation);
+  const dispatch = useDispatch();
+  const { data: latestConvoDetails, isLoading } = useGetConversationQuery(conversation?.id, {
```
Comment (can-aslan) [inline components/chat/ActiveWordsModal.tsx:21]: How would it be if this was isFetching like we did in other components?

File: components/chat/ActiveWordsModal.tsx [hunk 2]
```diff
@@ -5,44 +5,59 @@ import { useDispatch, useSelector } from 'react-redux';
...
-      dispatch(updateSelectedConversation({ conversation: updatedConvo }));
-    };
-    updateConvo();
-  }, [visible]);
+    if (latestConvoDetails) {
+      dispatch(updateSelectedConversation({ conversation: latestConvoDetails }));
+    }
+  }, [latestConvoDetails]);
+
+  if (isLoading) {
```
Comment (can-aslan) [inline components/chat/ActiveWordsModal.tsx:33]: How would it be if this was isFetching like we did in other components?

**Review by can-aslan — APPROVED** [review]
LGTM, nice implementation. Left some minor comments

File: components/chat/ActiveWordsModal.tsx (hunk 1 above)
Comment (selimcanglr) [reply components/chat/ActiveWordsModal.tsx:21]: I knowingly did it that way, because I don't want the loading screen to show up each time

**Pull request merged**